File: java/no_network_quiz/src/app/MultipleChoiceQuestions.java
package app;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

public class MultipleChoiceQuestions {
	private static final Path FILENAME = Paths.get("question.txt");

    private final QuestionAndAnswers m_questionData;

    public MultipleChoiceQuestions() throws IOException {
    	this(FILENAME);
    }

    public MultipleChoiceQuestions(Path a_filepath) throws IOException {
        m_questionData = loadQuestion(a_filepath);
    }

    public String getQuestion() {
        return m_questionData.getQuestion();
    }

    public String[] getShuffledAnswers() {
        return m_questionData.getShuffledAnswers();
    }

    public boolean isCorrectAnswer(String a_answer) {
        return m_questionData.isCorrectAnswer(a_answer);
    }

    private static QuestionAndAnswers loadQuestion(Path a_filepath) throws IOException {
        List<String> lines = Files.readAllLines(a_filepath);
        String question = lines.remove(0);
        String correctAnswer = lines.remove(0);
        String[] wrongAnswerOptions = new String[lines.size()];
        if (lines.get(lines.size() - 1).trim().isEmpty()) {
            lines.remove(lines.size() - 1);
        }
        lines.toArray(wrongAnswerOptions);
        return new QuestionAndAnswers(question, correctAnswer, wrongAnswerOptions);
    }
}

File: csharp/no_network_quiz/app/MultipleChoiceQuestions.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

public class MultipleChoiceQuestions
{
    private static readonly string FILENAME = "question.txt";

    private readonly QuestionAndAnswers m_questionData;

    public MultipleChoiceQuestions()
    {
        m_questionData = LoadQuestion(FILENAME);
    }

    public MultipleChoiceQuestions(string _filePath)
    {
        m_questionData = LoadQuestion(_filePath);
    }

    public string Question {
        get { return m_questionData.Question; }
    }

    public string[] GetShuffledAnswers()
    {
        return m_questionData.GetShuffledAnswers();
    }

    public bool IsCorrectAnswer(string _answer)
    {
        return m_questionData.IsCorrectAnswer(_answer);
    }

    private static QuestionAndAnswers LoadQuestion(string _filePath)
    {
        var lines = File.ReadAllLines(_filePath).ToList();
        string question = lines[0];
        lines.RemoveAt(0);
        string correctAnswer = lines[0];
        lines.RemoveAt(0);
        string[] wrongAnswerOptions = lines.ToArray();
        return new QuestionAndAnswers(question, correctAnswer, wrongAnswerOptions);
    }
}
